src/Swarm/TUI/View.hs

```diff
@@ -1097,6 +1098,20 @@ drawRobotLog s =
     (if i == n - 1 && s ^. uiState . uiScrollToEnd then visible else id) $
       drawLogEntry (not allMe) e
 
+-- | Show the 'CESK' machine of focused robot. Puts a separator above.
+drawRobotMachine :: AppState -> Bool -> Widget Name
+drawRobotMachine s showName = case s ^. gameState . to focusedRobot of
+  Nothing -> machineLine "no selected robot"
+  Just r ->
+    vBox
+      [ machineLine $ r ^. robotName <> "#" <> r ^. robotID . to tshow
+      , txt $ r ^. machine . to prettyText
```

Comment:
This doesn't wrap at all if the pretty-printed machine is very long, right?  How hard would it be to do that?  (Off the top of my head I forget.)

Reply:
Well... the REPL panel has a fixed size and this is part of it.

@byorgey do you mean that the REPL would have at least two lines, but would greedily fill the panel unless the Machine view takes it for itself?

I think we should improve the pretty printing of code in general so that it wraps nicely to multiple lines and we can know how much space is needed. If parts of the text are widgets (with colours) then we can also focus (with brick or manually) on the current expression, so it will always be in view.

Reply:
Good point, I forgot the REPL has a fixed size.  I don't know what I mean exactly, I just know that currently, if the CESK machine gets very big then it will just go off the right edge, and we should do something to improve it.

We should definitely improve the pretty printing of code in general, although that will not help with this particular issue since pretty-printing CESK machines and pretty-printing swarm-lang code use completely different code paths (for good reason, even though we try to use similar pretty-printed syntax as much as possible).